morph/cli.py

```diff
@@ -116,6 +131,109 @@ def cmd_count(args):
         print '%d\t%s' % (c, m.show().encode('utf-8'))
 
 
+def cmd_next(args):
+    notes_path = args.notes
+    prio_path = args.prio
+    mizer = MIZERS[args.mizer]
+
+    known = load_db('known')
+
+    candidates = defaultdict(list)
+    with codecs.open(notes_path, 'r', 'utf-8') as f:
+        for line in f.readlines():
+            note = line.strip()
+            text = note.split('\t', 1)[0]
+            unknowns = [m for m in mizer.getMorphemesFromExpr(text)
+                        if m not in known.db]
+            if len(unknowns) == 1:
+                candidates[unknowns[0].show()].append(note)
+
+    with codecs.open(prio_path, 'r', 'utf-8') as f:
+        for line in f.readlines():
+            freq, m = line.strip().split('\t', 1)
+            m_cite = m.split('\t', 1)[0]
+            for cand in candidates[m]:
+                print (u'%s\t%s\t%s' % (freq, m_cite, cand)).encode('utf-8')
+
+
+def cmd_grep(args):
+    pattern_string = args.pattern
+    files = args.files
+    max_count = args.max_count
+    mizer = MIZERS[args.mizer]
+
+    pattern = parse_morpheme(pattern_string.decode('utf-8'))
+
+    count = 0
+    for path in files:
+        with codecs.open(path, 'r', 'utf-8') as f:
+            for line in f:
+                ms = mizer.getMorphemesFromExpr(line)
+                if pattern in ms:
+                    sys.stdout.write(line.encode('utf-8'))
+                    count += 1
+                    if max_count is not None and count >= max_count:
+                        return
+
+
+def clear_locs(db, pred):
+    '''Remove from `db` locations that match `pred`, and morphemes left without locations.'''
+    empty_ms = []
+    for m, locs in db.db.iteritems():
+        locs.difference_update([loc for loc in locs if pred(loc)])
+        if not locs:
+            empty_ms.append(m)
+    for m in empty_ms:
+        del db.db[m]
+
+
+def cmd_sync_known(args):
+    filenames = args.input or ['known.txt']
+    should_merge = args.merge
+
+    external_db = load_db('external', allow_missing=True)
+    if not should_merge:
+        clear_locs(external_db, lambda loc: isinstance(loc, morph.morphemes.Nowhere))
```

Comment:
A new type is better suited. Maybe something like as name `SyncedExternal` (though this is heading in the direction of bikeshedding), which conveys the "sync" meaning you described. It should at least contain a little class-docstring which says that it gets deleted with these commands. I want to avoid following scenario in the (distant) future:

 - somebody contributes a new tool to put Nowhere locations inside the database; this seems with the current state of documentation like a good idea
 - this person crafts himself a beautiful external.db (many hours of hard work!)
 - this person finds your "sync" command and uses it to add the last morphemes from a text file
 - now all previously inserted locations and morphemes are gone 

> "Any existing morphemes in external.db which were previously inserted by this command are by default removed."

Yes, perfect. Short and precise.